### PR TITLE
feat: Add Any helper functions

### PR DESCRIPTION
**Pull request by amaurym**
This is an alternative solution to #91, it explores approach 1 defined in https://github.com/cosmos/cosmos-sdk/issues/13669.

Closes https://github.com/cosmos/cosmos-sdk/issues/13669

### CODE REVIEW

File: any/any.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+// message type if the intention is to unmarshal into it if the type matches:
+//
+//	m := new(foopb.MyMessage)
+//	if any.MessageIs(m) {
+//		if err := any.UnmarshalTo(m); err != nil {
+//			... // handle error
+//		}
+//		... // make use of m
+//	}
+//
```
Comment (tac0turtle) [inline any/any.go:118]: could we add a comment on what changed over the original version?

Comment (tac0turtle) [reply any/any.go:118]: oh now I see it in the doc.go

**Review by likhita-809 — APPROVED** [review]
lgtm

File: any/any.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// Code generated by protoc-gen-go. DO NOT EDIT.
```
Comment (aaronc) [inline any/any.go:31]: This code is not generated right?

Comment (amaurym) [reply any/any.go:31]: Google generated it into anypb, I copied it, and removed/updated stuff.

I can remove this comment though

Comment (aaronc) [reply any/any.go:31]: These are just helper functions not the Any type itself. I don't think all this stuff should be copied like this here

Comment (amaurym) [reply any/any.go]: Done. I still kept the license

**Comment by amaurym** [comment]
I'll merge this, as all reviews have been addressed

File: any/any_test.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	"github.com/cosmos/cosmos-proto/testpb"
+)
+
+func TestAny(t *testing.T) {
+	value := &testpb.A{SomeBoolean: true}
+
+	dst1 := &anypb.Any{}
+	err := any.MarshalFrom(dst1, value, proto.MarshalOptions{})
+	require.NoError(t, err)
+	require.Equal(t, "A", dst1.TypeUrl) // Make sure there's no "type.googleapis.com/" prefix.
```
Comment (amaurym) [inline any/any_test.go:22]: The type URL should be `/A` actually, to maintain compatibility with the `codec/types.Any`

**Pull request merged**